core/src/main/java/org/fao/geonet/kernel/IndexMetadataTask.java

```diff
@@ -125,6 +125,7 @@ public void run() {
             Log.error(Geonet.INDEX_ENGINE, "Error occurred indexing metadata", e);
         } finally {
             _batchIndex.remove(this);
+            _context.clearAsThreadLocal();
```

Comment:
To confirm no need to call `_context.clear()` as this code runs in a custom thread?

Reply:
This may be a case where a task is sharing a context, so calling clear() breaks another task.

Long term it may be smart to implement reference counting, or a context wrapper, when providing the same context to more than one thread.